### PR TITLE
LLVM 14 support

### PR DESCRIPTION
**Pull request by aykevl**
This PR adds support for LLVM 14 without dropping support for older versions. It switches the LLVM version for static builds, but doesn't switch the default for system LLVM builds yet because Homebrew doesn't have LLVM 14 yet.

This PR is similar to #2766 but is more complete. I ran a bunch of tests locally and it seems to work fine.

### CODE REVIEW

**Comment by deadprogram** [comment]
This can be rebased against `dev` now.

**Comment by deadprogram** [comment]
Pinging @QuLogic to take a look at this, if you please.

**Comment by aykevl** [comment]
Rebased and made a small change to update the wasi-libc build. Hopefully CI still passes...

File: cgo/libclang_config_llvm14.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+package cgo
+
+/*
+#cgo linux        CFLAGS:  -I/usr/lib/llvm-14/include
+#cgo darwin amd64 CFLAGS:  -I/usr/local/opt/llvm@14/include
+#cgo darwin arm64 CFLAGS:  -I/opt/homebrew/opt/llvm@14/include
+#cgo freebsd      CFLAGS:  -I/usr/local/llvm14/include
+#cgo linux        LDFLAGS: -L/usr/lib/llvm-14/lib -lclang
+#cgo darwin amd64 LDFLAGS: -L/usr/local/opt/llvm@14/lib -lclang -lffi
+#cgo darwin arm64 LDFLAGS: -L/opt/homebrew/opt/llvm@14/lib -lclang -lffi
```
Comment (QuLogic) [inline cgo/libclang_config_llvm14.go]: These are incorrect; cf #2608.

Comment (aykevl) [reply cgo/libclang_config_llvm14.go]: Thanks, fixed! My branch was rather old and so didn't have that fix yet.

**Comment by dkegel-fastly** [comment]
If you revert c89f02653231c5bbcaf7c0bbca30d5ef332e0eda on this branch, CI should run the test that was crashing, and see if #2762 is fixed by llvm14, I think.

**Comment by aykevl** [comment]
I think it's easier to merge this branch and then make a new PR with https://github.com/tinygo-org/tinygo/commit/c89f02653231c5bbcaf7c0bbca30d5ef332e0eda reverted.

**Comment by deadprogram** [comment]
This PR is the test https://github.com/tinygo-org/tinygo/pull/2793

**Comment by deadprogram** [comment]
Seems good to merge so we can work on the Windows issue without such a heavy rebuild.

Thanks @aykevl and @QuLogic now merging.

**Pull request merged**